src/hooks/useSwitchChain.ts

```diff
@@ -56,13 +53,13 @@ export default function useSwitchChain(): (chainId: SupportedChainId) => Promise
   const { connector, provider } = useWeb3React()
   const urlMap = useJsonRpcUrlsMap()
   return useCallback(
-    (chainId: SupportedChainId) => {
-      if (connector instanceof WalletConnect || connector instanceof Network) {
-        return connector.activate(chainId)
+    async (chainId: SupportedChainId) => {
+      try {
+        if (!provider) throw new Error()
+        await switchChain(provider, chainId, urlMap[chainId])
+      } catch {
+        await connector.activate(chainId)
```

Comment:
this could also fail though?

Reply:
If this fails, I'd expect it to be an error.